frame/democracy/src/lib.rs

```diff
@@ -313,7 +338,7 @@ decl_storage! {
 		pub PublicProps get(fn public_props): Vec<(PropIndex, T::Hash, T::AccountId)>;
 		/// Those who have locked a deposit.
 		pub DepositOf get(fn deposit_of):
-			map hasher(twox_64_concat) PropIndex => Option<(BalanceOf<T>, Vec<T::AccountId>)>;
+			map hasher(twox_64_concat) PropIndex => Option<(Vec<T::AccountId>, BalanceOf<T>)>;
```

Comment:
I don't understand why this is necessary or desirable. Feel free to elaborate or ignore.

Reply:
I'm happy to elaborate.

The complexity of copying and decoding the value in the runtime depends on the number of `seconds` (i.e. number of account in the vec). So to weight extrinsic using it I have to solution:
* find a hard limit.
* ask the user to provide an upper bound on this amount and check this upper bound before copying and decoding the full value into the runtime.

I choosed second option. But to check that the upper bound is correct I had to decode the Balance and then the len of the vec. But the type `BalanceOf<T>` doesn't say its encoded size so it is a bit painful to decode it. So I though: better to just change the order.
(But otherwise I could query the value by chunk of 512 bit until we have Balance decoded and then get the len.)

Reply:
This is actually quite problematic API wise since it needs to support old and new and everything in-between, especially since it is a Tuple. But will need figure out something.